evalml/pipelines/components/transformers/samplers/oversamplers.py

```diff
@@ -0,0 +1,133 @@
+from skopt.space import Integer
+
+from evalml.pipelines.components.transformers.samplers.base_sampler import (
+    BaseSampler
+)
+from evalml.utils import import_or_raise
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+class SMOTESampler(BaseSampler):
+    """SMOTE Oversampler component. Works on numerical datasets only. This component is only run during training and not during predict."""
+    name = "SMOTE Oversampler"
+    hyperparameter_ranges = {
+        'k_neighbors': Integer(2, 10)
+    }
+
+    def __init__(self, sampling_strategy='auto', k_neighbors=5, n_jobs=-1, random_seed=0, **kwargs):
```

Comment:
For other readers: @bchen1116 and I caught up and we're thinking of using the following API:
* `sampling_ratio`: a float in the range `(0, 1]` indicating the minority:majority ratio which the sampler should try to meet for every class (except the majority class). Default 0.25, aka a 1:4 ratio. This is the "simple" option to use oversampling.
* `sampling_ratio_dict`: a dictionary mapping each class value (which could be string, float, other) to a sampling ratio for that class. Default `None`. This is the "advanced" option to use oversampling.

@bchen1116 a "nice to have" is if the parameters we use for our undersampler match those used by the oversampler. Could we rename the undersampler ratio param to match, and also add a `sampling_ratio_dict` param to the undersampler? If you agree, I suggest you a) get a small separate PR up to rename the undersampler param and b) file an issue to track `sampling_ratio_dict` to undersampler.